test/puppetlabs/trapperkeeper/testutils/logging_test.clj

```diff
@@ -0,0 +1,150 @@
+(ns puppetlabs.trapperkeeper.testutils.logging-test
+  (:require
+   [clojure.test :refer :all]
+   [clojure.tools.logging :as log]
+   [puppetlabs.kitchensink.core :as kitchensink]
+   [puppetlabs.trapperkeeper.logging :refer [reset-logging root-logger-name]]
+   [puppetlabs.trapperkeeper.testutils.logging :as tgt :refer [event->map]])
+  (:import
+   [org.slf4j Logger LoggerFactory]))
+
+;; Without this, "lein test NAMESPACE" and :only invocations may fail.
+(use-fixtures :once (fn [f] (reset-logging) (f)))
+
+(deftest with-log-level-and-logging-to-atom
+  (let [expected {:logger "puppetlabs.trapperkeeper.testutils.logging-test"
+                  :level :info
+                  :message "wlta-test"
+                  :exception nil}]
+    (let [log (atom [])]
+      (tgt/with-log-level root-logger-name :error
+        (tgt/with-logging-to-atom root-logger-name log
+          (log/info "wlta-test"))
+        (is (not-any? #(= expected %) (map event->map @log)))))
+    (let [log (atom [])]
+      (tgt/with-log-level root-logger-name :info
+        (tgt/with-logging-to-atom root-logger-name log
+          (log/info "wlta-test"))
+        (is (some #(= expected %) (map event->map @log)))))))
+
+(def find-logger #'puppetlabs.trapperkeeper.testutils.logging/find-logger)
+(def log-event-listener #'puppetlabs.trapperkeeper.testutils.logging/log-event-listener)
```

Comment:
Well, I'm certainly not the expert, but I don't think there's any consensus on this front.  Even within our team, I believe opinions vary.  Personally, I mostly use private as a signal that "this thing isn't supported for external use".   Then, when I'm working in a namespace, aside from explicit circumventions like this (presumably only in testing code), I know that I should have substantial freedom to rework/remove anything private because no one else could have been depending on it (without an intentional sidestep).  And of course that's even more valuable when working on shared libraries.  In any case, I don't have a categorical opinion on the subject, just explaining my current approach.

(Mentioning @senior in case he wants to weigh in, since we've also discussed this from time to time.)